internal/gatewayserver/backendApi/core.go

```diff
@@ -0,0 +1,135 @@
+package backendapi
+
+import (
+	"context"
+
+	"github.com/fatih/structs"
+	"github.com/razorpay/trino-gateway/internal/gatewayserver/models"
+	"github.com/razorpay/trino-gateway/internal/gatewayserver/repo"
+)
+
+type Core struct {
+	backendRepo repo.IBackendRepo
+}
+
+type ICore interface {
```

Comment:
Have [this](https://github.com/razorpay/trino-gateway/pull/2/files#r805517660) same question here too.